keps/sig-node/3619-supplemental-groups-policy/README.md

```diff
@@ -790,13 +790,44 @@ rollout. Similarly, consider large clusters and how enablement/disablement
 will rollout across nodes.
 -->
 
+A rollout may fail when at least one of the following components are too old because this KEP introduces the new Kubernetes API field:
+
+| Component      | `supplementalGroupsPolicy` value that will cause an error |
+|----------------|-----------------------------------------------------------|
+| kube-apiserver | not null                                                  |
+| kubelet        | not null                                                  |
+| CRI runtime    | `Strict`                                                  |
+
+
+For example, an error will be returned like this if kube-apiserver is too old:
```

Comment:
I think you need to revise your version skew strategy (above) and think about how each component reacts during an upgrade. You can't say "kubelet must be at least the version of control-plane components". That's not realistic. It's not possible during an upgrade, and in fact people often run for extended periods of time with older kubelets. In that case, the kubelet won't see the new field. What sort of failure does that cause? Similarly for the CRI runtime.

Similarly, with enablement, you could enable the feature gate in the control plane, then only enable it in some nodes at the kubelet level. What's the behavior in this case?

If you enable it everywhere, then you create some pods with the `Strict` policy, then you disable it, will the kubelet see the new field or not? If it see it, have you feature gated the kubelet behavior to ignore the field?

Reply:
> You can't say "kubelet must be at least the version of control-plane components"

I did not say this. I think I respected [the version skew policy](https://kubernetes.io/releases/version-skew-policy/#kubelet).

> It's not possible during an upgrade, and in fact people often run for extended periods of time with older kubelets. In that case, the kubelet won't see the new field. What sort of failure does that cause?

This is a common issue when adding new API fields in Pod. For this KEP, the below matrix describes what will happen: 

| kubelet version | Feature Gate | CR support the KEP? | Pod's policy | Outcome |
|:--:|:--:|:--:|:--:|:--|
| <1.31<br/>(which does not know this field) | N/A | Yes/No | `Strict` | The pod can run, but its policy is just ignored. And, `.containerStatuses.user` will not be reported.|
| | | | `Merge`/(not set) | The pod can run normally as expected. And, `.containerStatuses.user` will not be reported. |
| >=1.31 | `True` | YES | `Strict` | The pod and its policy can run as expected. And, `.containerStatuses.user` will be reported. |
|  |  |  | `Merge`/(not set) | The pod and its policy can run as expected. And, `.containerStatuses.user` will be reported.|
|  |  | NO | `Strict` | The pod will be rejected in kubelet's admission. |
|  |  |  | `Merge`/(not set) | The pod and its policy can run as expected. And, `.containerStatuses.user` will be reported. |
| >=1.31 | `False` | YES  | `Strict` | The pod, which was created when the feature gate was enabled previously, and its policy can run as expected. But, `.containerStatuses.user` will not be reported.|
|  |  |  | `Merge`/(not set) | The pod and its policy can run as expected. But, `.containerStatuses.user` will not be reported.|
|  |  | NO | `Strict` | The pod, which was created when the feature gate was enabled previously, will be rejected in kubelet's admission.|
|  |  |  | `Merge`/(not set) | The pod and the policy can run as expected. But, `.containerStatuses.user` will not be reported.|

> Similarly, with enablement, you could enable the feature gate in the control plane, then only enable it in some nodes at the kubelet level. What's the behavior in this case?

Please see the above matrix.

> If you enable it everywhere, then you create some pods with the Strict policy, then you disable it, will the kubelet see the new field or not? If it see it, have you feature gated the kubelet behavior to ignore the field?

Please see the above matrix.

Reply:
That was a quote from line 676, please fix it up there and add in the matrix. Thanks!